tee-worker/ts-tests/common/utils.ts

```diff
@@ -397,3 +304,145 @@ export async function checkFailReason(
+        for (let i = 0; i < keyTypeIds.length; i++) {
+            const theKeyTypeId = keyTypeIds[i];
+            const theHasher = hashers[i].toString();
+            const theKeyItem = input[i];
+            // get the scale encoded input data by encoding the input
+            const theKeyType = metadata.registry.createLookupType(theKeyTypeId);
+            const theKeyItemEncoded = metadata.registry.createType(theKeyType, theKeyItem).toU8a();
+            // apply hasher
+            let theKeyItemAppliedHasher;
+            if (theHasher == 'Blake2_128Concat') {
+                theKeyItemAppliedHasher = blake2128Concat(theKeyItemEncoded);
+            } else if (theHasher == 'Twox64Concat') {
+                theKeyItemAppliedHasher = twox64Concat(theKeyItemEncoded);
+            } else if (theHasher == 'Identity') {
+                theKeyItemAppliedHasher = identity(theKeyItemEncoded);
+            } else {
+                throw new Error(`The hasher ${theHasher} is not support.`);
+            }
+            storageKey = u8aConcat(storageKey, theKeyItemAppliedHasher);
+        }
+    }
+    return storageKey;
+}
+export async function buildStorageData(
+    metadata: Metadata,
+    prefix: string,
+    method: string,
+    ...input: Array<unknown>
+): Promise<string | null> {
+    const storageEntry = buildStorageEntry(metadata, prefix, method);
+    if (!storageEntry) {
+        throw new Error('Can not find the storage entry from metadata');
+    }
+    let storageKey, valueType;
+
+    if (storageEntry.type.isPlain) {
+        storageKey = buildStorageKey(metadata, prefix, method);
+        valueType = metadata.registry.createLookupType(storageEntry.type.asPlain);
+    } else if (storageEntry.type.isMap) {
+        const { hashers, key, value } = storageEntry.type.asMap;
+        if (input.length != hashers.length) {
+            throw new Error('The `input` param is not correct');
+        }
+        storageKey = buildStorageKey(metadata, prefix, method, key, hashers, input);
+        valueType = metadata.registry.createLookupType(value);
+    } else {
+        throw new Error('Only support plain and map type');
+    }
+    console.debug(`storage key: ${u8aToHex(storageKey)}`);
+    return u8aToHex(storageKey);
+}
+
+export async function checkUserShieldingKeys(
+    context: IntegrationTestContext,
+    pallet: string,
+    method: string,
+    address: HexString
+): Promise<string> {
+    const storageKey = await buildStorageData(context.metaData, pallet, method, address);
+
+    let base58mrEnclave = base58.encode(Buffer.from(context.mrEnclave.slice(2), 'hex'));
+
+    let request = {
+        jsonrpc: '2.0',
+        method: 'state_getStorage',
+        params: [base58mrEnclave, storageKey],
+        id: 1,
+    };
+    let resp = await sendRequest(context.tee, request, context.api);
+
+    return resp.value;
+}
+export async function checkUserChallengeCode(
```

Comment:
A very small naming issue:
Why it has `User` in the name? If it has to match the storage method name, it should be without `User` and with `s`: `checkChallengeCodes`. If `User` simply means "user's", then the IDGraph should be named similarly, like `checkUserIDGraphs`